### PR TITLE
Add assertions and update ModelClassDiagram UML

### PR DESCRIPTION
**Pull request by LingZiyann**

### CODE REVIEW

File: src/main/java/seedu/address/model/tag/DifficultyTag.java [hunk 1]
```diff
@@ -37,6 +37,8 @@ public class DifficultyTag extends Tag {
      * @return true if the difficulty level is "low", "medium", or "high".
      */
     private static boolean isValidDifficulty(String difficultyLevel) {
+        assert difficultyLevel != null : "difficultyLevel should not be null";
```
Comment (vinceg4) [inline src/main/java/seedu/address/model/tag/DifficultyTag.java:40]: Great job on using assertions!

File: docs/diagrams/ModelClassDiagram.puml [hunk 2]
```diff
@@ -47,6 +53,12 @@ Company *--> CareerPageUrl
 Company *--> Address
 Company *--> "*" Tag
 
+Tag <|-- SalaryTag
```
Comment (vinceg4) [inline docs/diagrams/ModelClassDiagram.puml:56]: Good modelling of the different tags!

File: src/main/java/seedu/address/model/tag/PeriodTag.java [hunk 3]
```diff
@@ -47,6 +47,8 @@ public class PeriodTag extends Tag {
      * @return {@code true} if both the period and year are valid; {@code false} otherwise.
      */
     private static boolean isValidPeriod(String period, int year) {
+        assert period != null : "period should not be null";
```
Comment (vinceg4) [inline src/main/java/seedu/address/model/tag/PeriodTag.java:50]: Great job on using assertions!

File: docs/diagrams/ModelClassDiagram.puml (hunk 2 above)
Comment (mattisongjj) [reply docs/diagrams/ModelClassDiagram.puml:56]: This certainly makes the class diagram clearer and more accurate!

File: src/main/java/seedu/address/model/tag/DifficultyTag.java (hunk 1 above)
Comment (mattisongjj) [reply src/main/java/seedu/address/model/tag/DifficultyTag.java:40]: Could consider using the assert not null method from JUnit

File: src/main/java/seedu/address/model/tag/PeriodTag.java (hunk 3 above)
Comment (mattisongjj) [reply src/main/java/seedu/address/model/tag/PeriodTag.java:50]: Could consider using the assert not null method from JUnit

File: src/main/java/seedu/address/model/tag/SalaryTag.java [hunk 4]
```diff
@@ -38,6 +38,7 @@ public class SalaryTag extends Tag {
      * @return true if the Salary level is "low", "medium", or "high".
      */
     private static boolean isValidSalary(String salaryLevel) {
+        assert salaryLevel != null : "salaryLevel should not be null";
```
Comment (mattisongjj) [inline src/main/java/seedu/address/model/tag/SalaryTag.java:41]: Could consider using the assert not null method from JUnit

File: src/main/java/seedu/address/model/tag/WorkLifeBalanceTag.java [hunk 5]
```diff
@@ -39,6 +39,7 @@ public class WorkLifeBalanceTag extends Tag {
      * @return {@code true} if the input string is a valid level; {@code false} otherwise.
      */
     private static boolean isValidWorkLifeBalance(String workLifeBalanceLevel) {
+        assert workLifeBalanceLevel != null : "workLifeBalanceLevel should not be null";
```
Comment (mattisongjj) [inline src/main/java/seedu/address/model/tag/WorkLifeBalanceTag.java:42]: Could consider using the assert not null method from JUnit

**Comment by rynlwhh** [comment]
Agree with the rest comments, other than a few nits LGTM!

**Pull request merged**